File: kli_demo/employee-management/src/app/service/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, map } from 'rxjs';
import { EmployeeResponse } from '../model/interface/employee-response';
import { EmployeeFormRequest } from '../model/class/employee-form-request';

interface ApiResponse<T> {
  data: T;
  success: boolean;
  message: string;
}

@Injectable({ providedIn: 'root' })
export class EmployeeService {
  private baseUrl = 'http://localhost:8080/api/employees';

  constructor(private http: HttpClient) { }

  getAllEmployees(): Observable<EmployeeResponse[]> {
    return this.http.get<ApiResponse<EmployeeResponse[]>>(this.baseUrl)
      .pipe(map(res => res.data));
  }

  searchEmployees(search: string): Observable<EmployeeResponse[]> {
    return this.http.get<ApiResponse<EmployeeResponse[]>>(`${this.baseUrl}/search?search=${search}`)
      .pipe(map(res => res.data));
  }

  createEmployee(employee: EmployeeFormRequest): Observable<EmployeeResponse> {
    return this.http.post<ApiResponse<EmployeeResponse>>(this.baseUrl, employee)
      .pipe(map(res => res.data));
  }
}

File: employee-management/src/main/java/com/mindcraft/service/EmployeeService.java
package com.mindcraft.service;


import java.util.List;

import com.mindcraft.requestDTO.EmployeeRequest;
import com.mindcraft.responseDTO.ApiResponse;
import com.mindcraft.responseDTO.EmployeeResponse;

public interface EmployeeService {

    ApiResponse<EmployeeResponse> createEmployee(EmployeeRequest request);

    ApiResponse<List<EmployeeResponse>> getAllEmployees();

    ApiResponse<List<EmployeeResponse>> searchEmployees(String search);



}
